apps/namadillo/src/App/Sidebars/MaspRewardCalculator.tsx

```diff
@@ -0,0 +1,113 @@
+import { Panel, SkeletonLoading, StyledSelectBox } from "@namada/components";
+import { AssetImage } from "App/Transfer/AssetImage";
+import { maspRewardsAtom } from "atoms/chain";
+import clsx from "clsx";
+import { useAtomValue } from "jotai";
+import { useState } from "react";
+import { MaspAssetRewards } from "types";
+
+export const MaspRewardCalculator = (): JSX.Element => {
+  const rewards = useAtomValue(maspRewardsAtom);
+  const [amount, setAmount] = useState<string>("");
+  const [selectedAsset, setSelectedAsset] = useState<
+    MaspAssetRewards | undefined
+  >(rewards.data?.[0] || undefined);
+
+  const estimatedRewards =
```

Comment:
Unfortunately it's not that straightforward :/ At some point you have to call `simulateShieldedRewards` function othrwise the result will be incorrect.
Also it looks like `simulate_shielded_rewards` function in `sdk/mod.rs` is a bit bugged, but easy to fix:

```rs
+        let epoch = rpc::query_masp_epoch(self.namada.client()).await?;
+
         let (_, masp_value) = shielded
             .convert_namada_amount_to_masp(
                 self.namada.client(),
-                // Masp epoch should not matter
-                MaspEpoch::zero(),
+                epoch,
```

Reply:
Opened a [PR ](https://github.com/anoma/namada-sdkjs/pull/3)here. Thank you